include/rocksdb/secondary_cache.h

```diff
@@ -57,14 +57,31 @@ class SecondaryCache : public Customizable {
                                  std::shared_ptr<SecondaryCache>* result);
 
   // Insert the given value into this cache. The value is not written
-  // directly. Rather, the SaveToCallback provided by helper_cb will be
-  // used to extract the persistable data in value, which will be written
-  // to this tier. The implementation may or may not write it to cache
-  // depending on the admission control policy, even if the return status is
-  // success.
+  // directly. Rather, the SaveToCallback provided by helper will be
+  // used to extract value's persistable data (typically uncompressed block),
+  // which will be written to this tier. The implementation may or may not
+  // write it to cache depending on the admission control policy, even if the
+  // return status is success.
+  //
+  // If the implementation is asynchronous or otherwise uses `value` after
+  // the call returns, then Warm() must be overridden not to rely on Insert().
+  // For example, there could be a "holding area" in memory where Lookup might
+  // return the same parsed value back. But more typically, if the
+  // implementation only uses `value` for getting persistable data during the
+  // call, then the default implementation of `Warm()` suffices.
   virtual Status Insert(const Slice& key, void* value,
                         const Cache::CacheItemHelper* helper) = 0;
 
+  // Insert the given value from its persistable data (typically uncompressed
+  // block), as if generated by SaveToCallback/SizeCallback. This path is
+  // typically used in "warming up" the cache from some auxiliary source, so
+  // may or may not write it to cache depending on the admission control
+  // policy, even if the return status is success.
+  //
+  // The default implementation assumes synchronous, non-escaping Insert(),
+  // wherein `value` is not used after return of Insert(). See Insert().
+  virtual Status Warm(const Slice& key, const Slice& block);
```

Comment:
Not sure if `Warm` is the best name for it. Its more like an overload of `Insert`.

Reply:
From Google C++ style:
> You may overload a function when there are no semantic differences between variants.

I think the fact that one takes ownership of a value and has to use a callback to get persistable data makes it substantially different. And use of `void *` makes it hazardous to involve in overloading (e.g. accidentally passing Slice&, Slice*, and CacheItemHelper*)